TESTS/mbedmicro-rtos-mbed/malloc/main.cpp

```diff
@@ -21,7 +21,7 @@
   #error [NOT_SUPPORTED] test not supported
 #endif
 
-#define NUM_THREADS         5
```

Comment:
The large number of threads was specifically chosen to help this test catch race conditions faster and more reliably. When I was testing this with only 2 threads, race conditions wouldn't reliably get caught in the 15 seconds the test ran for. If you want to reduce the number of threads then you'll probably need to increase the test time.

You can verify this test is working by temporarily removing the malloc locks (__malloc_lock and __malloc_unlock) in platform/mbed_retarget.cpp and running this test and waiting for a failure. You should be able to run this test a couple of times with two threads to get the mean time to failure. From that you could then adjust the test time accordingly.

Reply:
Hi
I'm back...
Temp test done (removing malloc locks): either test is failing before first LED blink, either test is OK whatever the duration...

Reply:
Keep us updated (as we understand the test is failing?)

Reply:
New commit done.
Test report added.

Reply:
It can be fix in other way than thread number reduction, by removing extraneous heap allocations (by statically allocation of threads stack). 
We could go even further and put threads objects on stack


```
int8_t stack[THREAD_STACK_SIZE * NUM_THREADS];
bool thread_alloc_failure = false;
//Thread *thread_list[NUM_THREADS];
int test_time = 5;

// Allocate threads for the test
for (int i = 0; i < NUM_THREADS; i++) {
    thread_list[i] = new Thread(osPriorityNormal, THREAD_STACK_SIZE, stack + i * THREAD_STACK_SIZE);
```